pkg/codegen/python/utilities.go

```diff
@@ -109,10 +109,21 @@ var pypiDev = regexp.MustCompile("^dev[0-9]+$")
 // A valid post tag for pypi
 var pypiPost = regexp.MustCompile("^post[0-9]+$")
 
+// Transforms 0.0.1-alpha.18 to 0.0.1-alpha18; our users want to be able to use the previous form but semver.Version
+// parses it into two Pre segments ["alpha", "18"] which trips up translation. The same treatment is given beta and rc
```

Comment:
That's because it _is_ two segments. Semver defines pre-release as a dot separated identifiers where each individual identifier is either alphanumeric or just numeric. Which means "0.0.1-18.alpha" is a valid semver, but after your squashing here making it "0.0.1-18alpha" is now invalid and will trigger a panic in MustParse.

Reply:
Nope. `0.0.1-18alpha` does not match this regex. You can add a test. There's another bug in here not affected by my change that spits out suspicious version.

```
		{"0.0.1-18.alpha", "0.0.1+.alpha"},
```

Reply:
Ah right ok, missed the regex only checked one way. 
This conversion is fraught with issues in general 😆 but hey this covers one more case that's being used.

Reply:
Well, yes, I started comparing this conversion to the one used in pulumictl. It turns out that neither can pass the union of each other's test suites at the moment. Which his unfortunate. pulumictl is "more used in prod" as that currently decides versions for all pulumi packages including alphas and it does support the  v1.2.3-alpha.8 style out of the box..

Reply:
Yeh I dislike that pulumictl has that logic, it feels like it ought to be kept in just the language code generators but I get the oddity of publish workflows at the moment make that hard :(